lms/static/js/learner_dashboard/models/course_card_model.js

```diff
@@ -75,7 +75,7 @@ class CourseCardModel extends Backbone.Model {
     // These are raw course runs from the server. The start
     // dates are ISO-8601 formatted strings that need to be
     // prepped for display.
-    _.each(enrollableCourseRuns, (courseRun) => {
+    enrollableCourseRuns.forEach((courseRun) => {
```

Comment:
could this be done with Object assign or the spread operator?

```javascript
enrollableCourseRuns.forEach( courseRun =>
  Object.assign(courseRun, {
    start_date: CourseCardModel.formatDate(courseRun.start),
    end_date: CourseCardModel.formatDate(courseRun.end),
    dateString: this.formatDateString(courseRun),
  });
```

Reply:
@Mjloturco Why use [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)?

It looks like it will

> copy the values of all enumerable own properties from one or more source objects to a target object. It will return the target object.

The `forEach` just seems to be modifying the `courseRun`s. I think if you use `Object.assign` you'd want to reassign the target object to the `courseRun`?

```javascript
enrollableCourseRuns.forEach( courseRun => courseRun = Object.assign(courseRun, { thingsToModify }));
```

Reply:
@jaebradley The first object passed into assign is the target object.

```javascript 
enrollableCourseRuns.forEach( courseRun => courseRun = Object.assign({}, courseRun, { thingsToModify })); 
```
should be the same as
```javascript 
enrollableCourseRuns.forEach( courseRun => Object.assign(courseRun, { thingsToModify })); ```

Reply:
alternatively 
``` javascript
enrollableCourseRuns.forEach( courseRun => courseRun = { ...courseRun, { thingsToModify })); 
```

Reply:
@Mjloturco ah yes that's right!

![image](https://user-images.githubusercontent.com/8136030/37424994-49f9469a-2798-11e8-8743-ce0537e8735a.png)

- [RunKit](https://runkit.com/jaebradley/5aa9703fde545b0012efafbe)

Reply:
ha, could have shared https://codepen.io/mloturco/pen/bvpgqa

Reply:
Sure, I can do Object.assign. Looks a bit nicer.

I can't do `{ ...courseRun, { thingsToModify }}` because curiously, spread operators in object literals isn't supported until es2018. :man_shrugging: 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax